target/linux/ath79/image/generic-tp-link.mk

```diff
@@ -55,6 +55,18 @@ define Device/tplink_archer-c7-v2
 endef
 TARGET_DEVICES += tplink_archer-c7-v2
 
+define Device/tplink_archer-c7-v5
+  $(Device/tplink-safeloader-uimage)
+  ATH_SOC := qca9563
+  IMAGE_SIZE := 15104k
```

Comment:
The firmware partition differs a bit in size judging from your dtsi. The firmware partition is 0xf00000 (15360k) in size for the C7 v5.